### PR TITLE
Allow missing username and fix Docker build

### LINKED ISSUE #4056: Toil can't start when the current user has no username
As reported in https://github.com/ComparativeGenomicsToolkit/cactus/issues/677, the Kubernetes batch system calls `getpass.getuser()` when composing command line options, as Toil is starting up, but doesn't handle the `KeyError` it raises if you're running as a UID with no `/etc/passwd` entry and no username.

We should probably wrap that call, and any other requests for the current username, in a wrapper that always returns something even if no username is set at the system level.

┆Issue is synchronized with this [Jira Story](https://ucsc-cgl.atlassian.net/browse/TOIL-1150)
┆friendlyId: TOIL-1150

### PR DESCRIPTION
**Pull request by adamnovak**
This will fix #4056 and thus https://github.com/ComparativeGenomicsToolkit/cactus/issues/677, by handling the case where going to get the current user name throws an exception instead of producing a user name. In those cases, we will now use a fake user name string.

I had to revise the test system a bit to support running my new test file that I added to test this feature.

## Changelog Entry
To be copied to the [draft changelog](https://github.com/DataBiosphere/toil/wiki/Draft-Changelog) by merger:

 * Toil will no longer crash if the current user isn't in `/etc/passwd`. 
 * Toil Docker images now use Ubuntu 20.04 and Mesos 1.11.

## Reviewer Checklist

 * [ ] Make sure it is coming from `issues/XXXX-fix-the-thing` in the Toil repo, or from an external repo.
    * [ ] If it is coming from an external repo, make sure to pull it in for CI with:
        ```
        contrib/admin/test-pr otheruser theirbranchname issues/XXXX-fix-the-thing
        ```
    * [ ] If there is no associated issue, [create one](https://github.com/DataBiosphere/toil/issues/new).
* [ ] Read through the code changes. Make sure that it doesn't have:
    * [ ] Addition of trailing whitespace.
    * [ ] New variable or member names in `camelCase` that want to be in `snake_case`.
    * [ ] New functions without [type hints](https://docs.python.org/3/library/typing.html).
    * [ ] New functions or classes without informative docstrings.
    * [ ] Changes to semantics not reflected in the relevant docstrings.
    * [ ] New or changed command line options for Toil workflows that are not reflected in `docs/running/{cliOptions,cwl,wdl}.rst` 
    * [ ] New features without tests.
* [ ] Comment on the lines of code where problems exist with a review comment. You can shift-click the line numbers in the diff to select multiple lines.
* [ ] Finish the review with an overall description of your opinion.

## Merger Checklist

* [ ] Make sure the PR passes tests.
* [ ] Make sure the PR has been reviewed **since its last modification**. If not, review it.
* [ ] Merge with the Github "Squash and merge" feature.
    * [ ] If there are multiple authors' commits, add [Co-authored-by](https://github.blog/2018-01-29-commit-together-with-co-authors/) to give credit to all contributing authors.
* [ ] Copy its recommended changelog entry to the [Draft Changelog](https://github.com/DataBiosphere/toil/wiki/Draft-Changelog).
* [ ] Append the issue number in parentheses to the changelog entry.

### CODE REVIEW

**Review by w-gao — APPROVED** [review]
Looks great!

**Comment by adamnovak** [comment]
I'm adding some changes in here to fix  #4078.

**Comment by adamnovak** [comment]
OK, I managed to run a Mesos workflow with Mesos 1.11 and Ubuntu 20.04 with this. I haven't tested the autoscaling, but CI will.

File: src/toil/test/lib/test_misc.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+class UserNameUnvailableTest(ToilTest):
+    """
+    Make sure we can get something for a user name when user names are not
+    available.
+    """
+
+    def setUp(self):
+        super().setUp()
+        # Monkey pathch getpass.getuser to fail
```
Comment (w-gao) [inline src/toil/test/lib/test_misc.py:44]: `patch`

File: src/toil/test/lib/test_misc.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+class UserNameVeryBrokenTest(ToilTest):
+    """
+    Make sure we can get something for a user name when user name fetching is
+    broken in ways we did not expect.
+    """
+
+    def setUp(self):
+        super().setUp()
+        # Monkey pathch getpass.getuser to fail
```
Comment (w-gao) [inline src/toil/test/lib/test_misc.py:68]: `patch`

**Review by w-gao — COMMENTED** [review]
Looks like the autoscaling tests are passing as well!

Just noticed a small typo in the original PR but otherwise the new Mesos build seems to work great on Ubuntu 20.04.

I’m also wondering if there were other problems we had getting Mesos to work with later versions of Ubuntu before (slightly related to #3042), and if they are fixed now with newer versions of Mesos and Ubuntu?

File: src/toil/test/lib/test_misc.py (hunk 1 above)
Comment (adamnovak) [reply src/toil/test/lib/test_misc.py:44]:
```suggestion
        # Monkey patch getpass.getuser to fail
```

File: src/toil/test/lib/test_misc.py (hunk 2 above)
Comment (adamnovak) [reply src/toil/test/lib/test_misc.py:68]:
```suggestion
        # Monkey patch getpass.getuser to fail
```

**Pull request merged**